Routes/api/index.js

```diff
@@ -1,139 +1,157 @@
-import fs from "fs/promises";
-import path from "path";
-import express from "express";
+import fs from 'fs/promises';
+import path from 'path';
+import express from 'express';
+import cors from 'cors';
 let router = express.Router();
-import { config, getDirectoryEntries } from "@packages/common";
-const filesDir = path.resolve(import.meta.dirname, "../../", config.filesDir);
-const prvDir = path.resolve(import.meta.dirname, "../../", config.prvDir);
+import { config, getDirectoryEntries } from '@packages/common';
+const filesDir = path.resolve(import.meta.dirname, '../../', config.filesDir);
+const prvDir = path.resolve(import.meta.dirname, '../../', config.prvDir);
 import passport from 'passport';
 import passportHttp from 'passport-http';
-import fileUpload from "express-fileupload";
-import ipRangeCheck from "ip-range-check";
+import fileUpload from 'express-fileupload';
+import ipRangeCheck from 'ip-range-check';
 
 router.use((req, res, next) => {
-    next();
-})
+	next();
+});
 
 router.use('/', fileUpload());
 
+router.use(cors());
+
 router.use(async (req, res, next) => {
-    if (req.method != "GET") return next();
-    if (req.path.startsWith("/api/files")) return next();
-    try {
-        res.json(await getDirectoryEntries(decodeURIComponent(req.path).slice(1).split("/").slice(1).join("/")));
-    } catch (e) {
-        res.sendStatus(404);
-    }
-})
+	if (req.method != 'GET') return next();
+	if (req.path.startsWith('/api/files')) return next();
+	const offset = Number(req.query.offset) ?? 'x';
```

Comment:
`req.query.offset` が `undefined` なら `Number(req.query.offset)` は `NaN` を返す。
`null` や `undefined` を返すことはないはず。
だから `?? 'x'` の部分は不要だと思う。
```suggestion
	const offset = Number(req.query.offset);
```